pandas/tests/frame/test_stack_unstack.py

```diff
@@ -1175,6 +1175,30 @@ def test_stack_timezone_aware_values():
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize("dropna", [True, False])
+def test_stack_empty_frame(dropna):
+    # GH 36113
+    expected = Series(index=MultiIndex([[], []], [[], []]), dtype=np.float64)
+    result = DataFrame().stack(dropna=dropna)
+    tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("dropna", [True, False])
+@pytest.mark.parametrize("fill_value", [None, 0])
+def test_stack_unstack_empty_frame(dropna, fill_value):
+    # GH 36113
+    result = DataFrame().stack(dropna=dropna).unstack(fill_value=fill_value)
```

Comment:
This test fails in 32 bit.

Looks like also a problem with the dtype.
I am not sure if that is critical, but what if you specify here ``dtype=np.intp``?

```
>   def groupsort_indexer(const int64_t[:] index, Py_ssize_t ngroups):
E   ValueError: Buffer dtype mismatch, expected 'const int64_t' but got 'int'
```